release/release.go

```diff
@@ -105,15 +103,14 @@ func readOSRelease() OS {
 	return osRelease
 }
 
-var ioutilReadFile = ioutil.ReadFile
+// Note that osutil.FileExists cannot be used here as an osutil import will create a cyclic import
+var fileExists = func(path string) bool {
+	_, err := os.Stat(path)
+	return err == nil
+}
 
 func isWSL() bool {
-	version, err := ioutilReadFile("/proc/version")
-	if err == nil && bytes.Contains(version, []byte("Microsoft")) {
-		return true
-	}
-
-	return false
+	return fileExists("/proc/sys/fs/binfmt_misc/WSLInterop")
```

Comment:
Pardon my ignorance here, is this file always there regardless of WSL1,WSL2 custom kernels and all the possible variants that we may have?

Reply:
This file is in charge of inter-operating with Windows. It is there for WSL1, and standard WSL2. There is no 100% guarantee that it is going to be there for custom kernels as well, but at this point we'd be talking about something very different from standard WSL.

It's just more robust than kernel name, even if not fail-proof. Perhaps just as important, it is more consistent with other implementations in Ubuntu WSL.